WebView.Shared/Interop/WinRT/WebViewControlWebResourceRequestedEventArgs.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Security;
+using Windows.Web.Http;
+using windows = Windows;
+
+namespace Microsoft.Toolkit.Win32.UI.Controls.Interop.WinRT
+{
+    /// <summary>
+    /// Provides data for the <see cref="IWebView.WebResourceRequested" /> event. This class cannot be inherited.
+    /// </summary>
+    /// <remarks>Copy from <see cref="windows.Web.UI.WebViewControlWebResourceRequestedEventArgs"/> to avoid requirement to link Windows.winmd</remarks>
+    /// <seealso cref="System.EventArgs" />
+    /// <seealso cref="windows.Web.UI.WebViewControlWebResourceRequestedEventArgs"/>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly", MessageId = "WebResourceRequested", Justification = "Same as WinRT type")]
+    public sealed class WebViewControlWebResourceRequestedEventArgs : EventArgs
+    {
+        [SecurityCritical]
+        private readonly windows.Web.UI.WebViewControlWebResourceRequestedEventArgs _args;
+
+        internal WebViewControlWebResourceRequestedEventArgs(windows.Web.UI.WebViewControlWebResourceRequestedEventArgs args)
+        {
+            _args = args ?? throw new ArgumentNullException(nameof(args));
+        }
+
+        /// <summary>
+        /// Gets gets or sets the HTTP response that will be sent to the Windows.Web.UI.IWebViewControl
+        /// </summary>
+        /// <value>The response.</value>
+        public HttpResponseMessage Response => _args.Response;
```

Comment:
Can't leak this WinRT type--forces linking of WinRT SDK on consuming projects and creates a mess. For other instances that are supported by WebView we either have translators into native .NET types or omit the arguments from the event args. Examples include navigation where one can specify various HTTP options

Reply:
If it's too difficult or too much work to make translator/wrapper classes for `HttpRequestMessage` and `HttpResponseMessage`, I would be equally happy if the `WebViewControlWebResourceRequestedEventArgs.Request` and `Response` properties simply return `object` and leave it to the consumer to typecast these objects.  Although this is slightly unfriendly, I think it is reasonable to make WPF WebView slightly less friendly than UWP WebView because it is _already_ an unfriendly/unclean situation.  If someone wants it perfectly clean and friendly, then this is only possible if they fully switch their app to 100% UWP.  Whenever WPF (or WinForms) and UWP are mixed together in the same app, it's an unclean environment -- that's to be expected.

Thus the uncleanliness of `Request` and `Response` returning `object` is _nothing_ in comparison to the uncleanliness of mixing WPF and UWP together in the same app, therefore it's acceptable.

```suggestion
        public object Request => _args.Request;
        public object Response => _args.Response;
```

I would very much like to be able to use the `WebResourceRequested` event because we need access to the cookies via `WebViewControlWebResourceRequestedEventArgs.Request.Headers.Cookie` because we are required to download+display PDF files from websites that only authorize the PDF download if a particular temporary/session cookie is available, but WebView runs in a separate process, thus HTTP downloads in our app don't see the temporary cookies in the WebView process, thus we need the `WebResourceRequested` event to work around this problem.